bundles/org.openhab.binding.solarmax/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,202 @@
+			<channel id="softwareVersion" typeId="softwareVersion"/>
+			<channel id="buildNumber" typeId="buildNumber"/>
+			<channel id="startups" typeId="startups"/>
+			<channel id="acPhase1Current" typeId="acPhase1Current"/>
+			<channel id="acPhase2Current" typeId="acPhase2Current"/>
+			<channel id="acPhase3Current" typeId="acPhase3Current"/>
+			<channel id="energyGeneratedToday" typeId="energyGeneratedToday"/>
+			<channel id="energyGeneratedTotal" typeId="energyGeneratedTotal"/>
+			<channel id="operatingHours" typeId="operatingHours"/>
+			<channel id="energyGeneratedYesterday" typeId="energyGeneratedYesterday"/>
+			<channel id="energyGeneratedLastMonth" typeId="energyGeneratedLastMonth"/>
+			<channel id="energyGeneratedLastYear" typeId="energyGeneratedLastYear"/>
+			<channel id="energyGeneratedThisMonth" typeId="energyGeneratedThisMonth"/>
+			<channel id="energyGeneratedThisYear" typeId="energyGeneratedThisYear"/>
+			<channel id="currentPowerGenerated" typeId="currentPowerGenerated"/>
+			<channel id="acFrequency" typeId="acFrequency"/>
+			<channel id="acPhase1Voltage" typeId="acPhase1Voltage"/>
+			<channel id="acPhase2Voltage" typeId="acPhase2Voltage"/>
+			<channel id="acPhase3Voltage" typeId="acPhase3Voltage"/>
+			<channel id="heatSinkTemperature" typeId="heatSinkTemperature"/>
+		</channels>
+
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<label>Host</label>
+				<description>Hostname or IP Address</description>
+			</parameter>
+			<parameter name="portNumber" type="integer" required="false">
+				<label>Port</label>
+				<description>Port Number (defaults to 12345)</description>
+				<default>12345</default>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" required="false">
+				<label>Refresh Interval</label>
+				<description>Refresh Interval in seconds (defaults to 15)</description>
+				<default>15</default>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="lastUpdated">
+		<item-type>DateTime</item-type>
+		<label>Last Updated</label>
+		<description>Time when data was last read from the device</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="softwareVersion">
+		<item-type>Number</item-type>
+		<label>Software Version</label>
+		<description>Software Version installed on the SolarMax device</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="buildNumber">
+		<item-type>Number</item-type>
+		<label>Build Number</label>
+		<description>Firmware Build Number installed on the SolarMax device</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="startups">
+		<item-type>Number</item-type>
+		<label>Startups</label>
+		<description>Number of times the device has started</description>
+		<state readOnly="true"></state>
+	</channel-type>
+
+	<channel-type id="acPhase1Current">
+		<item-type>Number:ElectricCurrent</item-type>
```

Comment:
To make the Units of Measure work, you need to feed them with `QuantityType` instead of `DecimalType`.

Reply:
Not quite sure what change is being suggested here. A reference to the docs or an example would be great